code/__HELPERS/ray_helpers.dm

```diff
@@ -0,0 +1,585 @@
+/**
+  * Any unsorted global procs that help when locating turfs,  via
+  * casting rays out in specified directions etc
+  */
+
+/**
+  * Returns the turf offset from a given atom by the given dx and dy
+  * parameters
+  *
+  * This proc is bound to the map edges so will not overrun them
+  *
+  * Arguments:
+  * * atom/A the atom to calculate the offset from (the source)
+  * * dx, how many steps on the map grid to go in the x direction
+  * * dy, how many steps on the map grid to go in the y direction
+  */
+/proc/get_offset_target_turf(atom/A, dx, dy)
+	var/x = min(world.maxx, max(1, A.x + dx))
+	var/y = min(world.maxy, max(1, A.y + dy))
+	return locate(x,y,A.z)
```

Comment:
```suggestion
#define OFFSET_TARGET_TURF(source, dx, dy) (locate(min(world.maxx, max(1, source.x + dx), min(world.maxy, max(1, source.y + dy)), source.z))
```

Reply:
`clamp` is either just as fast or faster and more concise so I'd recommend using that.

Reply:
hey make your own PR

Reply:
Just accept the suggestion and do a ctrl+f replace `get_offset_target_turf` for `OFFSET_TARGET_TURF` ? The input parameters are the same, it doesn't require any added work.

Reply:
hey make your own PR

Reply:
:'(

Reply:
:<

Reply:
Alright, I'll open up a PR to change it once this is merged.